bindings/rust/s2n-tls/src/testing.rs

```diff
@@ -203,3 +214,15 @@ pub fn s2n_tls_pair(config: crate::raw::config::Config) {
 
     // TODO add assertions to make sure the handshake actually succeeded
 }
+
+struct MockClientHelloHandler {}
+
+impl ClientHelloHandler for MockClientHelloHandler {
+    fn poll_client_hello(
+        &mut self,
+        connection: &mut crate::raw::connection::Connection,
+    ) -> core::task::Poll<Result<(), ()>> {
+        connection.waker().unwrap().wake_by_ref();
```

Comment:
I think it would be a better test to return pending the first time and ready the second.

Reply:
I wanted to test that scenario but had trouble testing it.

- the cb fn only gets called once: https://github.com/aws/s2n-tls/blob/main/tls%2Fs2n_client_hello.c#L359-L361
- tried using a sleep future but also had to use `tokio_test::block_on` which defeated the purpose of the future.

Reply:
To simplify the Rust bindings and how Futures in Rust work, the `client_hello_callback` function should be polled (called more than once). This is being added in https://github.com/aws/s2n-tls/pull/3230